### DISCUSSION TITLE
enable setting annotated tag as default

### DISCUSSION THREAD

**Discussion by danmorgen** [gitextensions/gitextensions#10599] [Ideas]
Hi

When creating tag, the default option is lightweight tag.

Since there's no way in git to enforce annotated tags, I wish to change the default behavior in git extensions

Thanks

Dan

**Comment by vbjay** [comment]
My gpg work automatically handles this. 

https://github.com/gitextensions/gitextensions/pull/10473

In the meantime set tag.gpgSign to true and the user.signingKey default signing key.

**Comment by gerhardol** [comment]
GE is mostly sticking to Git defaults and the default for git-tag is lightweight tags.
There was a similar request in #9090 though.
Also #9672 tuned the defaults (not exactly the same though).
There are too many settings in GE already, I am OK with a PR to change the default to annotated.

**Reply by vbjay** [reply]
Once I get my work in...it will reflect the config value of tag.gpgSign. If they have that set...it will automatically select sign with default key. 

Part 2 of gpg work will be settings updates for git config to allow visibility of those settings.

**Reply by vbjay** [reply]
Try out the functionality

Pull my branch and run after setting
-  user.signingKey
- gpg.program
- tag.gpgSign
- commit.gpgSign

Play with gpgSign values to see how it reflects default functionality. 

You can use gpg4win to manage keys and create gpg keys. 

Try 
- commit window options menu
- create tag
- merge window
- Rebase showing dialog

See how it works.

**Reply by vbjay** [reply]
#10473

**Reply by vbjay** [reply]
Left to do is testing and rework of the controller work to make ui controller really a controller and not just a helper.
